truffle/src/main/java/org/jruby/truffle/nodes/RubyNode.java

```diff
@@ -214,13 +213,19 @@ public MemoryManager getMemoryManager() {
 
     // ruby() helper
 
-    protected Object ruby(VirtualFrame frame, String expression, Object... arguments) {
+    protected Object ruby(String expression, Object... arguments) {
+        CompilerAsserts.neverPartOfCompilation();
+        return ruby(Truffle.getRuntime().getCurrentFrame().getFrame(FrameInstance.FrameAccess.MATERIALIZE, true),
+                expression, arguments);
+    }
+
+    protected Object ruby(Frame frame, String expression, Object... arguments) {
         final MaterializedFrame evalFrame = setupFrame(frame, arguments);
         final DynamicObject binding = Layouts.BINDING.createBinding(getContext().getCoreLibrary().getBindingFactory(), evalFrame);
         return getContext().eval(expression, binding, true, "inline-ruby", this);
     }
 
-    private MaterializedFrame setupFrame(VirtualFrame frame, Object... arguments) {
+    private MaterializedFrame setupFrame(Frame frame, Object... arguments) {
         CompilerDirectives.transferToInterpreter();
         final MaterializedFrame evalFrame = Truffle.getRuntime().createMaterializedFrame(
```

Comment:
Are these changes needed or can be removed? Otherwise you'll get some conflict :wink: